src/core/Akka.Remote.Tests.MultiNode/RemoteNodeRestartGateSpec.cs

```diff
@@ -0,0 +1,177 @@
+                                });
+                    });
+                Sys.ActorSelection(new RootActorPath(secondAddress)/"user"/"subject").Tell("shutdown");
+            }, _specConfig.First);
+
+            RunOn(() =>
+            {
+                var addr = Sys.AsInstanceOf<ExtendedActorSystem>().Provider.DefaultAddress;
+                var firstAddress = Node(_specConfig.First).Address;
+
+                EnterBarrier("gated");
+
+                Sys.WhenTerminated.Wait(TimeSpan.FromSeconds(10));
+
+                var sb = new StringBuilder();
+                sb.AppendLine("akka.remote.retry-gate-closed-for = 0.5s")
+                    .AppendLine(" akka.remote.helios.tcp {")
+                    .AppendLine("hostname = " + addr.Host)
+                    .AppendLine("port =" + addr.Port)
+                    .AppendLine("}");
+
+                var freshSystem = ActorSystem.Create(Sys.Name, ConfigurationFactory.ParseString(sb.ToString())
+                    .WithFallback(Sys.Settings.Config));
+
+                var probe = CreateTestProbe(freshSystem);
+
+                // Pierce the gate
+                Within(TimeSpan.FromSeconds(30), () =>
+                {
+                    AwaitAssert(() =>
+
+                    {
+                        freshSystem.ActorSelection(new RootActorPath(firstAddress)/"user"/"subject")
+                            .Tell(new Identify("subject"), probe);
+                        probe.ExpectMsg<ActorIdentity>();
+                    });
+                });
+
+                // Now the other system will be able to pass, too
+                freshSystem.ActorOf(Props.Create(() => new Subject()), "subject");
+                Sys.WhenTerminated.Wait(TimeSpan.FromSeconds(30));
+            }, _specConfig.Second);
+        }
+
+        private class Subject : ActorBase
+        {
+            protected override bool Receive(object message)
+            {
+                if ("shutdown".Equals(message))
+                {
+                    Context.System.Terminate();
+                }
+                else
+                {
+                    Sender.Tell(message);
+                }
+
+                return true;
+            }
+        }
+    }
+
+    #endregion
+
+    #region Several different variations of the test
+
+    public class RemoteNodeRestartGateMultiNode1 : RemoteNodeRestartGateSpec
+    {
+    }
+
+    public class RemoteNodeRestartGateMultiNode2 : RemoteNodeRestartGateSpec
+    {
+    }
+
+    #endregion
+
+    #region Config
+
+    public class RemoteNodeRestartGateSpecConfig : MultiNodeConfig
+    {
+        public RemoteNodeRestartGateSpecConfig()
+        {
+            First = Role("first");
+            Second = Role("second");
+
+            CommonConfig = new Config(DebugConfig(false), ConfigurationFactory.ParseString(
+                @"akka.loglevel = DEBUG
+                   akka.remote.log-remote-lifecycle-events = INFO
```

Comment:
wrong value for  akka.remote.log-remote-lifecycle-events should be 'on' or 'off'